src/potions.ts

```diff
@@ -434,3 +430,121 @@ export function bathroomFinance(embezzlers: number): void {
     }
   }
 }
+
+class VariableMeatPotions {
+  potion: Item;
+  effect: Effect;
+  duration: number;
+  softcap: number; // Number of turns to cap out variable bonus
+  meatBonusPerTurn: number; // meat% bonus per turn
+  cappedMeatBonus: number;
+
+  constructor(
+    potion: Item,
+    softcap: number,
+    meatBonusPerTurn: number,
+    duration?: number,
+    effect?: Effect
+  ) {
+    this.potion = potion;
+    this.effect = effect ?? effectModifier(potion, "Effect");
+    this.duration = duration ?? numericModifier(potion, "Effect Duration");
+    this.softcap = softcap;
+    this.meatBonusPerTurn = meatBonusPerTurn;
+    this.cappedMeatBonus = softcap * meatBonusPerTurn;
+  }
+
+  use(quantity: number): boolean {
+    return use(quantity, this.potion);
+  }
+
+  price(historical: boolean): number {
+    // If asked for historical, and age < 14 days, use historical.
+    // If potion is not tradeable, use retrievePrice instead
+    return this.potion.tradeable
+      ? historical && historicalAge(this.potion) < 14
+        ? historicalPrice(this.potion)
+        : mallPrice(this.potion)
+      : retrievePrice(this.potion);
+  }
+}
+
+function triangleNumber(n2: number, n1?: number) {
```

Comment:
assign this a default value of 0 instead of using nullish coalescence

Reply:
also I think using `a` and `b` or something is less confusing than n1 and n2 but, again, very much a bachelor about it